examples/cpp/content_filter/SubscriberApp.cpp

```diff
@@ -0,0 +1,233 @@
+    if (config.filter_kind == CLIParser::FIlterKind::CUSTOM)
+    {
+        // Register the filter factory
+        if (eprosima::fastdds::dds::RETCODE_OK !=
+                participant_->register_content_filter_factory("MY_CUSTOM_FILTER", &filter_factory))
+        {
+            throw std::runtime_error("Custom filter initialization failed");
+        }
+    }
+
+    // Register the type
+    type_.register_type(participant_);
+
+    // Create the Subscriber
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+    if (subscriber_ == nullptr)
+    {
+        throw std::runtime_error("Subscriber initialization failed");
+    }
+
+    // Create the Topic
+    topic_ = participant_->create_topic(topic_name, type_.get_type_name(), TOPIC_QOS_DEFAULT);
+    if (topic_ == nullptr)
+    {
+        throw std::runtime_error("Topic initialization failed");
+    }
+
+    // Create the ContentFilteredTopic
+    std::string expression;
+    std::vector<std::string> parameters;
+    if (config.filter_kind == CLIParser::FIlterKind::CUSTOM)
+    {
+        // Custom filter: reject samples where index > parameters[0] and index < parameters[1].
+        // Custom filter does not use expression. However, an empty expression disables filtering, so some expression
+        // must be set.
+        expression = " ";
+        parameters.push_back(config.lower_bound);
+        parameters.push_back(config.upper_bound);
+        filter_topic_ =
+                participant_->create_contentfilteredtopic("HelloWorldFilteredTopic1", topic_, expression, parameters,
+                        "MY_CUSTOM_FILTER");
+    }
+    else if (config.filter_kind == CLIParser::FIlterKind::DEFAULT)
+    {
+        // Default filter: accept samples meeting the given expression: index between the two given parameters
+        expression = "index between %0 and %1";
+        parameters.push_back(config.lower_bound);
+        parameters.push_back(config.upper_bound);
+        filter_topic_ =
+                participant_->create_contentfilteredtopic("HelloWorldFilteredTopic1", topic_, expression, parameters);
+    }
+    else if (config.filter_kind == CLIParser::FIlterKind::NONE)
+    {
+        // An empty expression disables filtering
+        expression = "";
+        filter_topic_ =
+                participant_->create_contentfilteredtopic("HelloWorldFilteredTopic1", topic_, expression, parameters);
+    }
+
+    if (filter_topic_ == nullptr)
+    {
+        throw std::runtime_error("Filter Topic initialization failed");
+    }
+
+    // Create the DataReader
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    rqos.data_sharing().off();
+    // In order to receive all samples, DataReader is configured as RELIABLE and TRANSIENT_LOCAL (ensure reception even
+    // if DataReader matching is after DataWriter one)
+    rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    rqos.durability().kind = TRANSIENT_LOCAL_DURABILITY_QOS;
```

Comment:
It would be nice if we could these two into cli args and always set them on the tests